provider/alibabacloud/alibaba_cloud.go

```diff
@@ -373,24 +374,24 @@ func (p *AlibabaCloudProvider) groupRecords(records []alidns.Record) (endpointMa
 func (p *AlibabaCloudProvider) records() ([]alidns.Record, error) {
 	log.Infof("Retrieving Alibaba Cloud DNS Domain Records")
 	var results []alidns.Record
-
+	hostedZoneNames, err := p.getDomainList()
+	if err != nil {
+		log.Errorf("AlibabaCloudProvider getDomainList error %v", err)
+		return results, err
+	}
 	if len(p.domainFilter.Filters) == 0 {
-		domainNames, tmpErr := p.getDomainList()
-		if tmpErr != nil {
-			log.Errorf("AlibabaCloudProvider getDomainList error %v", tmpErr)
-			return results, tmpErr
-		}
-		for _, tmpDomainName := range domainNames {
-			tmpResults, err := p.getDomainRecords(tmpDomainName)
+		var tmpResults []alidns.Record
+		for _, zone := range hostedZoneNames {
+			tmpResults, err = p.getDomainRecords(zone)
 			if err != nil {
-				log.Errorf("AlibabaCloudProvider getDomainRecords %s error %v", tmpDomainName, err)
+				log.Errorf("AlibabaCloudProvider getDomainRecords %q error %v", zone, err)
```

Comment:
This should return an error to the caller, not log and ignore.

Reply:
in this loop we're getting records for all existing zones, if we fail to get records for one zone should we error out without adding the records from other zones?

Reply:
Yes, if you get a (probably transient) failure querying the cloud provider you should error out and make no changes rather than make changes based on incomplete data.

Reply:
makes sense, will make the changes